### PR TITLE
chore: use `rw.WriteHeader` to write responses without bodies

### PR DESCRIPTION
**Pull request by aslilac**
the body gets ignored anyway, so why bother allocating/serializing anything?

### CODE REVIEW

**Comment by Emyrk** [comment]
We also do this tracing span end too:
https://github.com/coder/coder/blob/41601183537884e6d75a73b58ad147d9e76f6258/coderd/httpapi/httpapi.go#L166-L167

We'd be losing that.

**Comment by aslilac** [comment]
yes, but all that span measures is time to encode the body and write it to the wire. I guess we could add a new function that just writes the head and tracks that with a span?

**Comment by Emyrk** [comment]
> _, span := tracing.StartSpan(ctx)

I misread what the span did, this is not a big deal :+1:

**Review by Emyrk — APPROVED** [review]

File: docs/api/users.md [hunk 1]
```diff
@@ -422,38 +421,11 @@ curl -X DELETE http://coder-server:8080/api/v2/users/{user} \
...
-```
-
 ### Responses
 
-| Status | Meaning                                                 | Description | Schema                                   |
-| ------ | ------------------------------------------------------- | ----------- | ---------------------------------------- |
-| 200    | [OK](https://tools.ietf.org/html/rfc7231#section-6.3.1) | OK          | [codersdk.User](schemas.md#codersdkuser) |
+| Status | Meaning                                                         | Description | Schema |
+| ------ | --------------------------------------------------------------- | ----------- | ------ |
+| 204    | [No Content](https://tools.ietf.org/html/rfc7231#section-6.3.5) | No Content  |        |
```
Comment (Emyrk) [inline docs/api/users.md:428]: This was a breaking change that we missed. I will put up a PR to revert.

**Pull request merged**